### ISSUE TITLE
bad parse on relative Uri

### ISSUE THREAD

**Issue by refried** [http4s/http4s#184]
The uri "g" gets parsed as the authority, but apparently it should be parsed as the path.

```
val jUri = new java.net.URI("g")
jUri: java.net.URI = g
scala> jUri.getAuthority
res7: String = null
scala> jUri.getPath
res8: String = g
```
vs
```
import org.http4s.Uri
val sUri = org.http4s.Uri.uri("g")
sUri: org.http4s.Uri = g/
scala> sUri.authority
res0: Option[org.http4s.Uri.Authority] = Some(g)
scala> sUri.path
res1: org.http4s.Uri.Path = /
```

This beastly regex copied from https://tools.ietf.org/html/rfc3986#appendix-B also parses "g" as the path:
```
val R = "^(([^:/?#]+):)?(//([^/?#]*))?([^?#]*)(\\?([^#]*))?(#(.*))?".r
val R(_,scheme,_,authority,path,_,query,_,fragment) = "g"
scheme: String = null
authority: String = null
path: String = g
query: String = null
fragment: String = null
```
I tried to convert this into a simple reference Uri parser, but got stuck on parsing the subcomponents of `Authority`.  Then I took a look at `Rfc3986Parser` but got scared.

**Comment by rossabaker** [comment]
java.net.URI is based on an older RFC, so I would not consider it authoritative.  The regex from the RFC makes a more compelling case that we have a problem.

**Comment by refried** [comment]
Oh yeah, I didn't even notice.  (RFC 2396 vs 3986)

**Comment by refried** [comment]
It looks like an authority has to have a `//` with it, i.e. `//g`.  
Even `foo.com/file` should all go into path, but `//foo.com/file` gets split into authority=`foo.com`, path=`/file`.  `http://foo.com/file` should get split into scheme, authority, and path like we're used to.

**Comment by bryce-anderson** [comment]
I believe this is closed via #191. @refried, if you discover differently, will you please reopen?

**Comment by refried** [comment]
LGTM
